ci/checks/check_min_airflow_dependency_compatibility.sh

```diff
@@ -0,0 +1,20 @@
+#!/bin/bash
+
+# Install great_expectations using the airflow constraints file for our minimum supported version
+
+AIRFLOW_VERSION=2.2.3
+PYTHON_VERSION="$(python --version | cut -d " " -f 2 | cut -d "." -f 1-2)"
+
+CONSTRAINT_URL="https://raw.githubusercontent.com/apache/airflow/constraints-${AIRFLOW_VERSION}/constraints-${PYTHON_VERSION}.txt"
+# 3.7 example
+# https://raw.githubusercontent.com/apache/airflow/constraints-2.2.3/constraints-3.7.txt
+
+echo "Install gx dependencies using airflow constraint file"
+echo "Python:${PYTHON_VERSION}\nAirflow:${AIRFLOW_VERSION}\n${CONSTRAINT_URL}\n"
+
+pip install -e . --constraint "${CONSTRAINT_URL}"
```

Comment:
Don't we also need to install using our constraint file? Can we pass both or merge the two first?

Reply:
Looks like we can pass both using multiple constraint flags https://pip.pypa.io/en/stable/cli/pip_install/#install-constraint

Reply:
I'll add it. Thanks.

Reply:
Thinking about this some more, I don't think we want to use our `constraints-dev.txt` as that's not how users install us. Doing this could potentially cause us to miss the dependency conflicts that our users experience.

Reply:
Good point - forgot that our constraints are only dev.